addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -185,15 +276,26 @@ options.registry.SocialMedia = options.Class.extend({
         }
         return true;
     },
+    /**
+     * Removes social media classes from the given element.
+     *
+     * @param  {HTMLElement} anchorEl
+     */
+    _removeSocialMediaClasses(anchorEl) {
+        let regx = new RegExp('\\b' + 's_social_media_' + '[^1-9][^ ]*[ ]?\\b', 'g');
+        anchorEl.className = anchorEl.className.replace(regx, '');
+        const iEl = anchorEl.querySelector('i');
+        regx = new RegExp('\\b' + 'fa-' + '[^1-9][^ ]*[ ]?\\b', 'g');
+        // Remove every fa classes except fa-x sizes.
+        iEl.className = iEl.className.replace(regx, '');
+    },
```

Comment:
You'll probably won't need this as a dedicated method after your refactoring -> again less spaguetti code. Spliting into methods is good... when the code needs to be reused. If you can have a code called at only one place and not create an extra method, it is often better (sometimes we do it anyway to allow custo / divide some large code but that's not a good case for that here).